src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -722,13 +728,19 @@ public override async ValueTask<FormulaValue> Visit(ResolvedObjectNode node, Eva
             }
         }
 
-        private FormulaValue GetVariableOrFail(ResolvedObjectNode node, ISymbolSlot slot)
+        private FormulaValue GetVariableOrFail(ResolvedObjectNode node, ISymbolSlot slot, bool isMutation = false)
         {
             if (_symbolValues != null)
             {
                 var value = _symbolValues.Get(slot);
                 if (value != null)
                 {
+                    if (isMutation)
+                    {
+                        value = value.MaybeShallowCopy();
+                        _symbolValues.Set(slot, value);

```

Comment:
This replaces in symbol table. 

But I think we need to modify the FormulaValue itself. 
If anybody else is holding onto the FormulaValue, they will have the stale result.

Reply:
this may also let us avoid the extra LazyNode for arg0.

Reply:
As discussed, yes this does replace the symbol table value, as designed.  This code will need to replace the outcome of the Patch in y, without impacting the existing value in x.
```
> Set( x, [1,2,3] )
x: [1, 2, 3]

> Set( y, x )
y: [1, 2, 3]

> Patch( y, {Value:2}, {Value:9} )
{Value:9}

> x
[1, 2, 3]

> y
[1, 9, 3]
```